components/controls/BoxedTabs.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Sonar UI Common
+ * Copyright (C) 2019-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as React from 'react';
+import { styled } from '../theme';
+
+export interface BoxedTabsProps<K> {
+  onSelect: (key: K) => void;
+  selected?: K;
+  tabs: Array<{ key: K; label: string }>;
+}
+
+const StyledTab = styled.button<{ active: boolean }>`
+  position: relative;
+  background-color: ${props => (props.active ? 'white' : props.theme.colors.barBackgroundColor)};
+  border: 1px solid ${props => props.theme.colors.barBorderColor};
+  border-right: none;
+  border-bottom: none;
```

Comment:
Maybe it makes sense to create a mixin for the border style and apply it to left & top rather than setting it for all border then to disable it for right and bottom.

Reply:
not a bad idea!